nautobot_design_builder/models.py

```diff
@@ -253,9 +253,26 @@ def delete(self, *args, **kwargs):
             raise ValidationError("A Design Instance can only be delete if it's Decommissioned and not Deployed.")
         return super().delete(*args, **kwargs)
 
+    def get_design_objects(self, model):
+        """Get all of the design objects for this design instance that are of `model` type.
+
+        For instance, do get all of the `dcim.Interface` objects for this design instance call
+        `design_instance.get_design_objects(Interface)`.
+
+        Args:
+            model (type): The model type to match.
+
+        Returns:
+            Queryset of matching objects.
+        """
+        entries = JournalEntry.objects.filter_by_instance(self, model=model)
+        return model.objects.filter(pk__in=entries.values_list("pk", flat=True))
+
     @property
     def created_by(self):
         """Get the username of the user who created the object."""
+        # TODO: if we just add a "created_by" and "last_updated_by" field, doesn't that
+        # reduce the complexity of code that we have in the util module?
```

Comment:
not following, what do you want to reduce?

Reply:
I think this is a note for me to think about in the future. We have a property to lookup the `created_by` information, but I was thinking we could just store that directly when the design is implemented instead of trying to do complex queries at a later time.